scripts/github_ci.py

```diff
@@ -13,18 +13,78 @@
 
 import json
 from collections.abc import Sequence
-from typing import Annotated, cast
+from typing import Annotated, Literal, cast
 
 import rich
 import typer
+from rich import pretty
 
 from . import _common as common, nox_sessions
 
 
 cli = typer.Typer(no_args_is_help=True, name="github-ci", help=__doc__)
 
 
-def create_github_actions_list(
+def collect_affected_sessions(
+    *,
+    config: str = nox_sessions.DEFAULT_CONFIG,
+    base_commit: str | None = None,
+    verbose: bool = False,
+) -> tuple[list[str], list[str]]:
+    """Collect affected test sessions based on changes from a base commit."""
+
+    config_path = config.format(REPO_ROOT=common.REPO_ROOT)
+    sessions = nox_sessions.load_test_sessions_config(config_path)
+    if verbose:
+        rich.print(
+            f"Found {len(sessions)} test sessions in {config_path}: {[session['name'] for session in sessions]}"
+        )
+
+    collect_all = base_commit is None
+    all_session_names = [session["name"] for session in sessions]
+    affected = (
+        all_session_names
+        if collect_all
+        else [
+            session["name"]
+            for session in sessions
+            if nox_sessions.should_run_session(
+                session, base_commit=cast(str, base_commit), verbose=verbose
+            )
+        ]
+    )
+
+    if verbose:
+        rich.print(
+            f"Found {len(affected)} test sessions in changes from {base_commit!r}: {affected}"
+        )
+
+    return affected, all_session_names
+
+
+def compute_output(
+    sessions: Sequence[str],
+    kind: Literal["affected", "excluded"],
```

Comment:
Do you want to keep both? Since there are no tests, there is the danger of the unused one to break. Or maybe we want to extract this into a package that could be used in other projects, e.g. icon4py.